### PR TITLE
Make DatetimeIndex timezone naive for compatibility with xarray/numpy engine

### LINKED ISSUE #1052: timezones make time index of wrong type
Reproducible example:
```python
import pandas as pd
from darts import TimeSeries

data = {
    "val1": {
        pd.Timestamp("2020-09-08 00:00:00+0200", tz="CET", freq="H"): 0.0,
        pd.Timestamp("2020-09-08 01:00:00+0200", tz="CET", freq="H"): 0.0,
        pd.Timestamp("2020-09-08 02:00:00+0200", tz="CET", freq="H"): 0.0,
        pd.Timestamp("2020-09-08 03:00:00+0200", tz="CET", freq="H"): 0.0,
        pd.Timestamp("2020-09-08 04:00:00+0200", tz="CET", freq="H"): 0.0,
    }
}

df = pd.DataFrame.from_dict(data)
X = TimeSeries.from_dataframe(df, value_cols=None)
df_utc = df.tz_convert("UTC").tz_localize(None)
X_utc = TimeSeries.from_dataframe(df_utc, value_cols=None)
```

It seems the types of time index of `X` and `X_utc` are different. For example doing `X_utc.plot()` works, but `X.plot()` complains:
```
TypeError: Plotting requires coordinates to be numeric, boolean, or dates of type numpy.datetime64, datetime.datetime, cftime.datetime or pandas.Interval. Received data of type object instead.
```

### PR DESCRIPTION
**Pull request by madtoinou**
Fixes #1052 and #648. Xarray relies on numpy engine, which is not timezone aware. Causes issues for plotting and resampling.

### Summary
When constructing a `TimeSeries` from a `pd.DataFrame` or a `pd.Series` with a `DatetimeIndex` containing timezone information, this attribute is dropped without conversion using `tz_localize(None)` (conversion can causes shift in the timestamps, making the index invalid). The timezone information can be recovered by calling `ts.time_index.tz_localize(original_tz)` prior to exporting the data.

### Other Information
The `DatetimeIndex` not used as index are silently converted to UTC by the `pd.DataFrame` constructor without saving the timezone information. Thus, timezone aware dataset should use the time range as index to take advantage of the bugfix and avoid the error raised by the `TimeSeries` constructor due to index invalidity.

In order to have the desired x-axis when plotting the `TimeSeries` (with a timezone unaware `time_index` due to the bugfix), the user can simply leverage the matplotlib.pyplot `rcParams['timezone']` parameter to automatically convert the time axis back to the original timezone.

### CODE REVIEW

File: darts/timeseries.py [hunk 1]
```diff
@@ -654,7 +654,17 @@ def from_dataframe(
...
                 logger,
             )
-            time_index = df.index
+            # BUGFIX : force time-index to be timezone naive as xarray doesn't support it
+            # pandas.DataFrame loses the tz information if it's not its index
+            if isinstance(df.index, pd.DatetimeIndex) and df.index.tz is not None:
+                print(
+                    "The provided DatetimeIndex was associated with a timezone, which is currently not supported "
+                    "by xarray. To avoid unexpected behaviour, the tz information was removed, don't forget to "
+                    f"call `ts.time_index.tz_localize({df.index.tz})` when exporting the results."
```
Comment (hrzn) [inline darts/timeseries.py:663]:
```suggestion
                    f"`ts.time_index.tz_localize({df.index.tz})` when exporting the results."
```

**Review by hrzn — COMMENTED** [review]
Looks good! I just have suggestions for slightly better warning messages, and also to remove the `print()` statements.

File: darts/timeseries.py [hunk 2]
```diff
@@ -831,6 +841,14 @@ def from_series(
         TimeSeries
             A univariate and deterministic TimeSeries constructed from the inputs.
         """
+        # BUGFIX : force time-index to be timezone naive as xarray doesn't support it
+        if isinstance(pd_series, pd.DatetimeIndex) and pd_series.dt.tz is not None:
```
[the remark this answers is not in the document]
Comment (madtoinou) [reply darts/timeseries.py:845]: You're right, it's redundant. And the `.index` is indeed missing.

**Review by hrzn — APPROVED** [review]
Thanks 👍

**Pull request merged**